fonz/connection.py

```diff
@@ -0,0 +1,172 @@
+        self.client = None
+        self.headers = None  # type: Optional[JsonDict]
+
+        logging.info('Instantiated Fonz object for url: {}'.format(url))
+
+    def connect(self) -> None:
+        """Authenticate, start a dev session, check out specified branch."""
+
+        logging.info('Authenticating Looker credentials.')
+
+        login = requests.post(
+            url=compose_url(self.url, 'login'),
+            data={
+                'client_id': self.client_id,
+                'client_secret': self.client_secret
+                })
+
+        access_token = login.json()['access_token']
+        self.headers = {'Authorization': 'token {}'.format(access_token)}
+
+    def update_session(self) -> None:
+
+        logging.info('Updating session to use development workspace.')
+
+        update_session = requests.patch(
+            url=compose_url(self.url, 'session'),
+            headers=self.headers,
+            json={'workspace_id': 'dev'})
+
+        logging.info('Setting git branch to: {}'.format(self.branch))
+
+        update_branch = requests.put(
+            url=compose_url(self.url, 'projects', self.project, 'git_branch'),
+            headers=self.headers,
+            json={'name': self.branch})
+
+    def get_explores(self) -> List[JsonDict]:
+        """Get all explores from the LookmlModel endpoint."""
+
+        logging.info('Getting all explores in Looker instance.')
+
+        models = requests.get(
+            url=compose_url(self.url, 'lookml_models'),
+            headers=self.headers)
+
+        explores = []
+
+        logging.info('Filtering explores for project: {}'.format(self.project))
+
+        for model in models.json():
+            if model['project_name'] == self.project:
+                for explore in model['explores']:
+                    explores.append({
+                        'model': model['name'],
+                        'explore': explore['name']
+                        })
+
+        return explores
+
+    def get_explore_dimensions(self, explore: JsonDict) -> List[str]:
+        """Get dimensions for an explore from the LookmlModel endpoint."""
+
+        logging.info('Getting dimensions for {}'.format(explore['explore']))
+
+        lookml_explore = requests.get(
+            url=compose_url(
+                self.url,
+                'lookml_models',
+                explore['model'],
+                'explores',
+                explore['explore']),
+            headers=self.headers)
+
+        dimensions = []
+
+        for dimension in lookml_explore.json()['fields']['dimensions']:
+            dimensions.append(dimension['name'])
+
+        return dimensions
+
+    def get_dimensions(self, explores: List[JsonDict]) -> List[JsonDict]:
+        """Finds the dimensions for all explores"""
+        for explore in explores:
+            explore['dimensions'] = self.get_explore_dimensions(explore)
+
+        return explores
+
+    def create_query(self, explore: JsonDict) -> int:
```

Comment:
That's definitely feasible in certain places, we just would need something that handles the flow of data.

There are some instances where I think it will be less feasible though. For example, to get dimensions, you need to pass it both the model and the explore, which is why I passed around the `List[JsonDict]`. I don't know if there's a better way around that?